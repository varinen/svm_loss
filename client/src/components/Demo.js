import React from 'react';
import Background from './Background';
import Visualization from './Visualization';
import Params from './Params';
import Datapoints from './Datapoints';

const Demo = () => (
            <div className="content-wrapper">
                <Background/>
                <Visualization />
                <Params />
                <Datapoints />
            </div>
        );

export default Demo;