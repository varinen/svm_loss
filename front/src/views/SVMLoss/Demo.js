import React, {Fragment} from 'react';
import {Grid, Typography} from "@material-ui/core";

import Background from "./Demo/Background";
import Visio from './Demo/Visio';
import Params from './Demo/Params';
import Data from "./Demo/Data";


export default function Demo(props) {
    console.log('Demo props', props);
    return (
        <Fragment>
            <Grid container spacing={3} alignItems="stretch">
                <Grid item xs={12}>
                    <Background/>
                </Grid>

                <Grid item container spacing={2}>
                    <Grid item md={7} sm={12}>
                        <Visio store={props.store}/>
                    </Grid>
                    <Grid item md={5} sm={12} container>
                        <Grid item xs container direction="column" alignItems="stretch">
                            <Grid item xs>
                                <Params store={props.store}/>
                            </Grid>
                            <Grid item xs>
                                <Data  store={props.store}/>
                            </Grid>
                        </Grid>
                    </Grid>
                </Grid>

                <Grid item xs={12}>
                    <Typography variant="subtitle1" gutterBottom>
                        Copyright Â© SVM Loss 2019
                    </Typography>
                </Grid>
            </Grid>
        </Fragment>
    );
}