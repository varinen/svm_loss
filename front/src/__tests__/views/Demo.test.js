import React from 'react';
import {shallow} from 'enzyme';
import Demo from '../../views/SVMLoss/Demo';

describe('Demo', () => {
    const component = shallow(<Demo/>);

    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
    });
});

