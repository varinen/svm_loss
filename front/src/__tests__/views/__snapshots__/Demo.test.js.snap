// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo renders correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    alignItems="stretch"
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <_default />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={7}
        sm={12}
      >
        <Visio />
        <Connect(WithStyles(Hyper)) />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        container={true}
        item={true}
        md={5}
        sm={12}
      >
        <WithStyles(ForwardRef(Grid))
          alignItems="stretch"
          container={true}
          direction="column"
          item={true}
          xs={true}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={true}
          >
            <Params />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={true}
          >
            <Data />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        gutterBottom={true}
        variant="subtitle1"
      >
        Copyright Â© SVM Loss 2019
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
